### PR TITLE
Add DNS provider for Servercow.

### LINKED ISSUE #990: Add DNS Support for servercow
Hi,
I have added the DNS Support for servercow some time ago to acme.sh (https://github.com/Neilpang/acme.sh/blob/master/dnsapi/dns_servercow.sh).

Could you please add this to lego?

I would try to do it by myself, but I am not firm with golang :(

https://cp.servercow.de/client/plugin/support_manager/knowledgebase/view/34/dns-api-v1/7/

### PR DESCRIPTION
**Pull request by ldez**
Add DNS provider for Servercow.

https://cp.servercow.de/client/plugin/support_manager/knowledgebase/view/34/dns-api-v1/7/

Fixes #990

### CODE REVIEW

**Comment by ldez** [comment]
from https://github.com/go-acme/lego/issues/990#issuecomment-583867713

```
2020/02/09 17:54:36 [INFO] acme: Registering account for xxx@xxx.xxx
!!!! HEADS UP !!!!

                Your account credentials have been saved in your Let's Encrypt
                configuration directory at "/home/jens/lego/.lego/accounts".
                You should make a secure backup of this folder now. This
                configuration directory will also contain certificates and
                private keys obtained from Let's Encrypt so making regular
                backups of this folder is ideal.2020/02/09 17:54:37 [INFO] [*.daenerys-targaryen.de, daenerys-targaryen.de] acme: Obtaining bundled SAN certificate
2020/02/09 17:54:37 [INFO] [*.daenerys-targaryen.de] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/38021096
2020/02/09 17:54:37 [INFO] [daenerys-targaryen.de] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/38021097
2020/02/09 17:54:37 [INFO] [*.daenerys-targaryen.de] acme: use dns-01 solver
2020/02/09 17:54:37 [INFO] [daenerys-targaryen.de] acme: Could not find solver for: tls-alpn-01
2020/02/09 17:54:37 [INFO] [daenerys-targaryen.de] acme: Could not find solver for: http-01
2020/02/09 17:54:37 [INFO] [daenerys-targaryen.de] acme: use dns-01 solver
2020/02/09 17:54:37 [INFO] [*.daenerys-targaryen.de] acme: Preparing to solve DNS-01
2020/02/09 17:54:39 [INFO] [daenerys-targaryen.de] acme: Preparing to solve DNS-01
2020/02/09 17:54:41 [INFO] [*.daenerys-targaryen.de] acme: Trying to solve DNS-01
2020/02/09 17:54:41 [INFO] [*.daenerys-targaryen.de] acme: Checking DNS record propagation using [208.67.222.222:53 208.67.220.220:53 [2001:4860:4860::8888]:53 [2001:4860:4860::8844]:53]
2020/02/09 17:54:41 [INFO] Wait for propagation [timeout: 1m0s, interval: 2s]
2020/02/09 17:54:46 [INFO] [*.daenerys-targaryen.de] The server validated our request
2020/02/09 17:54:46 [INFO] [daenerys-targaryen.de] acme: Trying to solve DNS-01
2020/02/09 17:54:46 [INFO] [daenerys-targaryen.de] acme: Checking DNS record propagation using [208.67.222.222:53 208.67.220.220:53 [2001:4860:4860::8888]:53 [2001:4860:4860::8844]:53]
2020/02/09 17:54:46 [INFO] Wait for propagation [timeout: 1m0s, interval: 2s]
2020/02/09 17:54:47 [INFO] [daenerys-targaryen.de] The server validated our request
2020/02/09 17:54:47 [INFO] [*.daenerys-targaryen.de] acme: Cleaning DNS-01 challenge
2020/02/09 17:54:49 [INFO] [daenerys-targaryen.de] acme: Cleaning DNS-01 challenge
2020/02/09 17:54:52 [INFO] [*.daenerys-targaryen.de, daenerys-targaryen.de] acme: Validations succeeded; requesting certificates
2020/02/09 17:54:54 [INFO] [*.daenerys-targaryen.de] Server responded with a certificate.
```

File: providers/dns/servercow/servercow.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	config.Username = values["SERVERCOW_USERNAME"]
+	config.Password = values["SERVERCOW_PASSWORD"]
+
+	return NewDNSProviderConfig(config)
+}
+
+// NewDNSProviderConfig return a DNSProvider instance configured for Servercow.
+func NewDNSProviderConfig(config *Config) (*DNSProvider, error) {
+	if config.Username == "" || config.Password == "" {
+		return nil, fmt.Errorf("servercow: some credentials information are missing")
```
Comment (dmke) [inline providers/dns/servercow/servercow.go]: The error message is a bit vague, better "servercow: incomplete credentials, missing username/password"?

**Comment by ldez** [comment]
Currently, I prefer `defer func() { _ = something.Close() }()` due to the explicit ignore of the error.
It's a bit more verbose but it's mush easy to read and check ignored errors.

**Comment by dmke** [comment]
OK, thanks.

File: .travis.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ matrix:
     - go: 1.x
     - go: tip
   allow_failures:
+    - go: 1.x # FIXME currently golangci-lint doesn't work with go1.14
```
Comment (dmke) [inline .travis.yml]: Is 1.14 already released? The [release notes](https://golang.org/doc/devel/release.html) doesn't list it yet...

Comment (ldez) [reply .travis.yml]: nope but Travis update the version behind 1.x

https://travis-ci.com/go-acme/lego/jobs/291075495#L230

Comment (ldez) [reply .travis.yml]: Based on 1.12 the release will come shortly :wink:

Comment (dmke) [reply .travis.yml]: Yeah, both the go1.14 tag already exist (https://github.com/golang/go/tree/go1.14), and 1.14 is listed on the [download page](https://golang.org/dl/#stable).

**Pull request merged**